modules/javafx.media/src/main/native/jfxmedia/platform/osx/avf/AVFMediaPlayer.mm

```diff
@@ -803,8 +803,16 @@ - (BOOL)resourceLoader:(AVAssetResourceLoader *)resourceLoader
         }
 
         NSMutableData* readData = nil;
+        bool isRandomAccess = locatorStream->GetCallbacks()->IsRandomAccess();
```

Comment:
`URIConnectionHolder` always return `false` for "jar" or "jrt" protocol. Why it was added?

Reply:
Yes, you right `MemoryConnectionHolder` in case of jar, but only for AudioClip. MediaPlayer does not use it. I tested AudioClip with jar with and without random access. Without random access I only hear some clicking sounds, so looks like it does not work completely. With random access I can only hear audio for like 30 seconds and then it loops, so looks like it is broken as well.

Can you play audio file correctly via AudioClip?

Reply:
I figure it out. Seek is ignored for random access, so setting `pos = 0` will always read from begging of the file. We should read from `dataRequest.requestedOffset`. I am attaching proposed patch which I tested and works fine.